analyzers/tests/SonarAnalyzer.UnitTest/TestCases/LockedFieldPrivateReadonly.cs

```diff
@@ -0,0 +1,180 @@
+﻿using System;
+using System.Linq;
+
+class Test
+{
+    private static readonly object staticReadonlyField = null;
+    private static object staticReadWriteField = null;
+
+    private readonly object readonlyField = null;
+    private readonly string readonlyStringField = "a string";
+    private object readWriteField = null;
+
+    private static object StaticReadonlyProperty => null;
+    private object ReadonlyProperty => null;
+
+    private static object StaticReadWriteProperty { get; set; }
+    private object ReadWriteProperty { get; set; }
+
+    void OnAStaticField()
+    {
+        lock (staticReadonlyField) { }
+        lock (staticReadWriteField) { }                  // Noncompliant {{'staticReadWriteField' is not 'private readonly', and should not be used for locking.}}
+        //    ^^^^^^^^^^^^^^^^^^^^
+        lock (Test.staticReadonlyField) { }
+        lock (Test.staticReadWriteField) { }             // Noncompliant {{'staticReadWriteField' is not 'private readonly', and should not be used for locking.}}
+        //    ^^^^^^^^^^^^^^^^^^^^^^^^^
+        lock (AnotherClass.staticReadonlyField) { }      // Noncompliant {{'staticReadonlyField' is not 'private readonly', and should not be used for locking.}}
```

Comment:
I don't know about this one. It doesn't seem dangerous to me to lock on a `readonly` field that's not `private`. 
It's more about locking on a field that's not declared by the same class. I'm not sure if this should be an issue. At the very least I'd rephrase the message to reflect that (e.g. "Use fields from 'Test' for locking.")

Reply:
My reasoning for this scenario would be the following: if a (`static`) field is `readonly`, but not `private` to `AnotherClass`, any external code could lock it, interfering with the synchronization of methods in `AnotherClass`. I think that's the reason why the rule mentions `private` in its title, along with `readonly`.

I agree that, in this scenario, making the field `private` would make it disappear from the visibility scope of `Test`, and that `Use fields from 'Test' for locking` is a better message.

I have added code to detect the specific scenario of "field of another class" and rephrase in the way you suggested.